usng2-js/build.gradle.kts

```diff
@@ -72,25 +79,75 @@ tasks {
         // Setting the args below is necessary if using a scoped package
         // setArgs(listOf("--access", "public"))
     }
+
+    "installBrowserify"(NpmTask::class) {
+        dependsOn("npm-index")
+        setNpmCommand("install")
+        setArgs(listOf("browserify"))
+    }
+    "installGCC"(NpmTask::class) {
+        dependsOn("npm-index")
+        setNpmCommand("install")
+        setArgs(listOf("google-closure-compiler"))
+    }
+
+    "browserifyIt"(NodeTask::class) {
+        dependsOn("installBrowserify", "webpack-bundle", "build")
+        setScript(file("node_modules/.bin/browserify"))
+        setArgs(listOf("$buildDir/js/usng2-js.js",
+                "--standalone",
+                "usng2",
+                "--outfile",
+                "$buildDir/bundle/browserified.js"))
+    }
+    "gccIt"(NodeTask::class) {
+        dependsOn("browserifyIt", "installGCC")
+        setScript(file("node_modules/.bin/google-closure-compiler"))
+        setArgs(listOf("--js",
+                "${project.buildDir}/bundle/browserified.js",
+                "--js_output_file",
+                "${project.buildDir}/bundle/gcc-output.js"))
+    }
 }
 
 open class NpmPrePublish : DefaultTask() {
     @TaskAction
     fun run() {
         val outDir = project.mkdir("${project.buildDir}/npmpub_tmp")
-        project.copy {
-            from("${project.rootDir}/README.md") {
-                rename("README", "readme")
+
+        copyMetaFiles(outDir)
+        copyJs(outDir)
+
+        editAndCopyPackageJson(outDir)
+    }
+
+    fun copyMetaFiles(outDir: File) {
+        arrayOf("README", "LICENSE").forEach {
+            project.copy {
+                from("${project.rootDir}/$it.md") {
+                    rename(it, it.toLowerCase())
+                }
+                into(outDir)
             }
-            into(outDir)
         }
+    }
+
+    private fun copyJs(outDir: File) {
         project.copy {
-            from("${project.buildDir}/package.json")
+            from("${project.buildDir}/bundle/gcc-output.js") {
+                rename("gcc-output", "usng2.main")
```

Comment:
✏️ Would it be worth moving the final name into a variable?

Reply:
👍